docs/register-bitcoin-stake.md

```diff
@@ -0,0 +1,698 @@
+      // the signature follows encoding in either BIP-340 spec or BIP-322 spec
+      bytes btc_sig = 2;
+  }
+  ```
+* `btc_pk`:
+  This Bitcoin `secp256k1` public key of the BTC staker,
+  in BIP-340 format (Schnorr signatures). It is a compact, 32-byte
+  value derived from the staker's private key. This public key corresponds to
+  the staker public used to construct the [staking script](./staking-script.md) used
+  in the BTC Staking transaction.
+* `fp_btc_pk_list`:
+  A list of the `secp256k1` public keys of the finality providers
+  (FPs) to which the stake is delegated in BIP-340 format (Schnorr signatures)
+  and compact 32-byte representation.
+  **For phase-2,
+  this list contains a single key,
+  since Babylon is the only system secured by the Bitcoin stake**.
+  The public key should be exactly the same as the one
+  used when constructing the [staking script](./staking-script.md).
+* `staking_time`:
+  The duration of staking in Bitcoin blocks. This is the same
+  as the timelock used when constructing the [staking script](./staking-script.md)
+  and must comply with the Babylon staking parameters.
+* `staking_value`:
+  The amount of satoshis locked in the staking output of the BTC
+  staking transaction (`staking_tx`). This value must precisely match the Bitcoin
+  amount in the staking transaction.
+* `staking_tx`:
+  The Bitcoin staking transaction in hex format. This transaction locks
+  the Bitcoin funds in the staking output and it is crucial that it follows
+  precisely the [staking script](./staking-script.md) format using the
+  correct Bitcoin staking parameters. Some of the values included in the
+  `MsgCreateBTCDelegation` should match with the values in the staking transaction.
+  > **⚠️ Important: Should the Inputs of the Staking Transaction be Signed?**
+  >
+  > Whether the staking transaction should include signed or unsigned
+  > UTXO inputs depends on the type of inputs used.
+  > Since the Babylon chain tracks staking transactions based on
+  > their transaction hash, it's crucial for the staker to submit
+  > the staking transaction in a format that contains all the
+  > necessary data to generate the full hash corresponding
+  > to the fully signed transaction.
+  > 
+  > For legacy inputs that require the `script_sig` field to be filled,
+  > the signature **must** be included when submitting the
+  > transaction to Babylon, as this field directly influences
+  > the transaction hash.
+  > 
+  > On the other hand, for inputs that require the `witness` field
+  > (e.g., SegWit inputs), the signature **does not** need to be
+  > included as part of the transaction.
+  > 
+  > While this distinction doesn't affect the guarantees of staking
+  > transactions following the post-staking registration flow
+  > (since stakes are first committed to Bitcoin), stakers opting
+  > for the pre-staking registration flow using only legacy inputs
```

Comment:
I think I would reword this a bit to indicate that even if there is only one lagacy input, it must be submitted with the signature.

So lets say we have staker who created staking transaction with 4 inputs. Lets say 1 of them is legacy. This means staker should submit staking transaciton in which:
- 3 inputs are provided without witness (to minimize the costs)
- 1 inputs is provided with script sig filled